src/matcher/expression.js

```diff
@@ -106,14 +98,86 @@ const checkAlternative = match => {
   const altIndexes = []
   const regex = runRegexCheck(
     match,
-    RX_ALT,
+    new RegExp(RX_ALT, 'g'),
     // Use a non-capture group to allow matching, but don't include in the results (?:)
-    (val, ...args) => `(?:${val.trim().replace('/', '|')})`
+    (val, ...args) => `(?:${val.trim().replace(/\//g, '|')})`
   )
 
   return { regex, altIndexes }
 }
 
+/**
+ * Adds regex anchors to the ends of the regex string, if it needs them
+ * @param {string} str
+ * @return {string} with anchors
+ */
+const checkAnchors = str => {
```

Comment:
This seems odd that you would need to add these?
* It's seems this would cause more issues due to
  * white space issues
  * matching text starting or ending with a expression
* I can see from the code, that the step match text is converted Regex is before calling this, so it should not be an issue. 
  * But that just begs the question; why do you need to add this as well?

Reply:
I went back into my unit tests and found the case that breaks without them: any case in which the feature text includes all the right step definition patterns but also adds more text at the start or end.

Example:
* Step Definition: `"I have apples to eat"`
* Feature step text: `"I have apples to eat and throw"`
* That should **not** match, unless I am mistaken. Without the anchors, though, it will in fact match.
* The feature step text "John and I have apples to eat" would also match

Reply:
Ahh, that makes perfect sense. Thank you for the clarification.